### PR TITLE
Add Proxywide Message Relay

### PR DESCRIPTION
**Pull request by mmis1000**

### CODE REVIEW

**Comment by sciuridae0603** [comment]
@OKTW-Network/frontend

File: src/main/kotlin/one/oktw/galaxy/proxy/event/ChatExchange.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import com.velocitypowered.api.proxy.ServerConnection
+import net.kyori.text.TextComponent
+import net.kyori.text.serializer.ComponentSerializers
+import one.oktw.galaxy.proxy.Main
+import one.oktw.galaxy.proxy.api.ProxyAPI
+import one.oktw.galaxy.proxy.model.ChatData
+import java.util.*
+
+class ChatExchange(val topic: String) {
+    val eventId = "galaxy/chat"
```
Comment (james58899) [inline src/main/kotlin/one/oktw/galaxy/proxy/event/ChatExchange.kt]:
```suggestion
    val eventId = MinecraftChannelIdentifier.create("galaxy", "proxy/chat")
```

File: src/main/kotlin/one/oktw/galaxy/proxy/event/ChatExchange.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Subscribe
+    fun onServerSend(event: PluginMessageEvent) {
+        if (event.identifier.id != eventId) return
```
Comment (james58899) [inline src/main/kotlin/one/oktw/galaxy/proxy/event/ChatExchange.kt]:
```suggestion
        if (!event.identifier.equals(eventId)) return
```

File: api/src/main/kotlin/one/oktw/galaxy/proxy/api/packet/MessageSendResponse.kt [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+data class MessageSendResponse(
+    // default to uuid connection associated with
+    val sender: UUID = ProxyAPI.dummyUUID,
+
+    // message id
+    val id: UUID = ProxyAPI.dummyUUID,
+    // result for this operation
+    val result: Boolean = true
+) : Packet {
+    override val type = PacketTypes.MessageSend
```
Comment (james58899) [inline api/src/main/kotlin/one/oktw/galaxy/proxy/api/packet/MessageSendResponse.kt]:
```suggestion
    override val type = PacketTypes.MessageSendResponse
```

File: src/main/kotlin/one/oktw/galaxy/proxy/event/ChatExchange.kt [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+class ChatExchange(val topic: String) {
+    companion object {
+        val eventId = MinecraftChannelIdentifier.create("galaxy", "proxy/chat")
+        val eventIdResponse = MinecraftChannelIdentifier.create("galaxy", "proxy/chat-response")
+    }
+
+    var MESSAGE_TIMEOUT = 2000L
+    val topicResponse = "$topic-response"
+    val dummyUUID = UUID.fromString("00000000-0000-0000-0000-000000000000")
```
[the remark this answers is not in the document]
Comment (mmis1000) [reply src/main/kotlin/one/oktw/galaxy/proxy/event/ChatExchange.kt]: 那是 internal field

[the remark this answers is not in the document]
Comment (james58899) [reply src/main/kotlin/one/oktw/galaxy/proxy/event/ChatExchange.kt]: 讓他 public

**Pull request merged**